agent/reconciler/host_reconciler.go

```diff
@@ -28,14 +29,26 @@ import (
 	infrastructurev1beta1 "github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/apis/infrastructure/v1beta1"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate . Installer
+type Installer interface {
+	Install(string, string, string, logr.Logger) error
+	Uninstall(string, string, string, logr.Logger) error
+}
```

Comment:
We could rethink on creating an exportable `Installer Interface` in the `installer` package. There are not going to be multiple implementations but still it can provide value in testing and readability. thoughts?? @anusha94 @jamiemonserrate

Reply:
Yes, you're right. We need to rethink a few things on the installer package. But maybe not part of writing tests? These unit / integration tests have been deferred forever :D

Reply:
> provide value in testing and readability

True. The initial take was the installer will be exposed only via call to `New` and everything else should be accessed by the struct that is returned to facilitate data hiding (I think).

So maybe we don't want to change that behavior as part of this PR and take it up when we potentially add more features to the installer. Does that sound okay?

Reply:
> So maybe we don't want to change that behavior as part of this PR and take it up when we potentially add more features to the installer. Does that sound okay?

yup definitely not in this PR, but just to keep in mind for future refactoring.